lib/ui/worker/cache.dart

```diff
@@ -660,6 +662,40 @@ class CacheEntry {
 
   /// Byte data of this [CacheEntry].
   final Uint8List? bytes;
+
+  /// [MediaType] type of this [CacheEntry].
+  Future<MediaType?> get type async {
+    if (!_typeResolved) {
+      _type = await _resolveType();
+    }
+    return _type;
+  }
+
+  /// Value of [MediaType] type of this [CacheEntry].
+  MediaType? _type;
+
+  /// True if [_resolveType] was called.
+  bool _typeResolved = false;
```

Comment:
[if null is a valid initialized value for the variable, then it probably does make sense to have a separate boolean field.](https://dart.dev/effective-dart/usage#avoid-late-variables-if-you-need-to-check-whether-they-are-initialized)